framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1523,6 +1522,39 @@ private void handleExpr(AnnotationBuilder builder, String name, Expression expr)
                                 + " and expected: "
                                 + expected);
             }
+        } else if (expr instanceof ClassExpr) {
+            ClassExpr classExpr = (ClassExpr) expr;
+            ExecutableElement var = builder.findElement(name);
+            TypeMirror expected = var.getReturnType();
+            if (expected.getKind() == TypeKind.DECLARED) {
+                String className = classExpr.getType().toString();
+                Class<?> clazz;
+                try {
+                    // If it is a simply-named class literal then fully-qualified class
+                    // name should be found in stubfile imports.
+                    for (ImportDeclaration impDecl : theCompilationUnit.getImports()) {
+                        if (impDecl.getName().getIdentifier().equals(className)) {
+                            className = impDecl.getNameAsString();
+                        }
+                    }
+                    clazz = Class.forName(className);
+                } catch (ClassNotFoundException e) {
+                    ErrorReporter.errorAbort("StubParser: unknown class name " + className);
+                    throw new Error("dead code; this can't happen");
+                }
+                TypeMirror val =
+                        TypesUtils.typeFromClass(
+                                atypeFactory.getContext().getTypeUtils(),
+                                atypeFactory.getElementUtils(),
+                                clazz);
+                builder.setValue(name, val);
+            } else {
```

Comment:
The case for boolean expressions above also handles arrays.
Does this handle an annotation `Class<?>[] value()` and one passes a single class literal?